src/main/java/org/junitpioneer/jupiter/StdOutputStream.java

```diff
@@ -10,34 +10,34 @@
 
 package org.junitpioneer.jupiter;
 
+import java.io.ByteArrayOutputStream;
 import java.io.OutputStream;
-import java.io.StringWriter;
 import java.nio.charset.Charset;
 import java.util.Arrays;
 
 abstract class StdOutputStream extends OutputStream {
 
-	private final StringWriter writer = new StringWriter();
+	private final ByteArrayOutputStream out = new ByteArrayOutputStream();
 
 	public StdOutputStream() {
 		// recreate default constructor to prevent compiler warning
 	}
 
 	@Override
 	public void write(int i) {
-		writer.write(i);
+		out.write(i);
 	}
 
 	@Override
 	public final void write(byte[] b, int off, int len) {
-		writer.write(new String(b, Charset.defaultCharset()), off, len);
+		out.write(b, off, len);
 	}
 
 	/**
 	 * @return the string that was written to {@code System.out} or {@code System.err}
 	 */
 	public String capturedString() {
-		return writer.toString();
+		return out.toString(Charset.defaultCharset());
```

Comment:
Is having a `Charset.defaultCharset()` necessary?

Reply:
You are right, it is not. The original code had it while constructing the string where it was necessary but here it is not.
```suggestion
		return out.toString();
```